exo-setup/src/app-setup.spec.ls

```diff
@@ -0,0 +1,123 @@
+      fs.exists @docker-compose-location, (exists) ~>
+        expect(exists).to.be.true
+        @docker-compose = yaml.safe-load fs.read-file-sync(@docker-compose-location, 'utf8')
+        done!
+
+    specify 'should list all services and dependencies under \'services\'', (done) ~>
+      expect(Object.keys @docker-compose.services).to.eql([@exocom-name, @service-name, @mongo-name])
+      done!
+
+    specify 'should generate an image name for each dependency', (done) ~>
+      expect(@docker-compose.services[@exocom-name].image).to.not.be.empty
+      expect(@docker-compose.services[@mongo-name].image).to.not.be.empty
+      done!
+
+    specify 'should generate a container name for each service/dependency equal to the service/dependency name', (done) ~>
+      expect(@docker-compose.services[@service-name].container_name).to.eql(@service-name)
+      expect(@docker-compose.services[@mongo-name].container_name).to.eql(@mongo-name)
+      expect(@docker-compose.services[@exocom-name].container_name).to.eql(@exocom-name)
+      done!
+
+    specify 'should have the correct command for each service/dependency that can be built', (done) ~>
+      expect(@docker-compose.services[@service-name].command).to.eql('node_modules/exoservice/bin/exo-js')
+      expect(@docker-compose.services[@exocom-name].command).to.eql('bin/exocom')
+      done!
+
+    specify 'should have correct build path for each service', (done) ~>
+      expect(@docker-compose.services[@service-name].build).to.eql('../mongo')
+      done!
+
+    specify 'should include \'exocom\' in the dependencies of every service', (done) ~>
+      expect(@docker-compose.services[@service-name].depends_on).to.include(@exocom-name)
+      done!
+
+    specify 'should include external dependencies', (done) ~>
+      expect(@docker-compose.services[@service-name].depends_on).to.include(@mongo-name)
+      done!
+
+    specify 'should set up an \'environment\' for exocom with a port and correct service routes', (done) ~>
+      exocom-env = @docker-compose.services[@exocom-name].environment
+      expect(exocom-env.ROLE).to.eql('exocom')
+      expect(exocom-env.PORT).to.eql('$EXOCOM_PORT')
+      expect(exocom-env.SERVICE_ROUTES).to.eql('[{"role":"mongo","receives":null,"sends":null}]')
+      done!
+
+    specify 'should set up an \'environment\' for each service with exocom as host', (done) ~>
+      mongo-env = @docker-compose.services[@service-name].environment
+      expect(mongo-env.ROLE).to.eql(@service-name)
+      expect(mongo-env.EXOCOM_HOST).to.eql(@exocom-name)
+      expect(mongo-env.EXOCOM_PORT).to.eql('$EXOCOM_PORT')
+      done!
+
+    specify 'should generate a volume path for an external dependency that mounts a volume', (done) ~>
+      expect(@docker-compose.services[@mongo-name].volumes).to.not.be.empty
+      done!
+
+  # indirectly tests @cat-file used in exosphere-shared/src/compile-service-routes.ls
+  describe 'app with external docker images' (...) ->
```

Comment:
Once we switch to the more complex example app, this test case will be covered and can be removed. We can test `@cat-file` in another PR if we decide to do so.